inc/class-component.php

```diff
@@ -378,80 +378,119 @@ public function get_theme_options(): array {
 	}
 
 	/**
-	 * Add one or more theme options.
+	 * Test if a string is in the theme options.
 	 *
-	 * @param array|string $themes One or more themes to add.
+	 * @param string $theme Theme name.
+	 * @return bool
+	 */
+	public function is_available_theme( string $theme ): bool {
+		return in_array( $theme, $this->get_theme_options(), true );
+	}
+
+	/**
+	 * Set theme options.
+	 *
+	 * @param array $theme_options New theme options. Ensures uniques.
 	 * @return self
 	 */
-	public function add_theme_options( $themes ): self {
+	public function set_theme_options( array $theme_options ): self {
+		$this->theme_options = array_unique( $theme_options );
+		return $this;
+	}
 
-		// Convert to array if necessary.
-		if ( is_string( $themes ) ) {
-			$themes = [ $themes ];
-		}
+	/**
+	 * Add theme options.
+	 *
+	 * @param array $theme_options One or more theme names to add.
+	 * @return self
+	 */
+	public function add_theme_options( array $theme_options ): self {
 
-		// Merge the new value(s).
-		$this->theme_options = array_merge(
-			$this->theme_options,
-			$themes
-		);
+		array_map(
+			function( $theme_option ) {
+
+				// Ignore non-strings for now.
+				if ( ! is_string( $theme_option ) ) {
+					return;
+				}
 
-		// Sanitize the entire thing.
-		$this->sanitize_theme_options();
+				$this->add_theme_option( $theme_option );
+			},
+			$theme_options
+		);
 
 		return $this;
 	}
 
 	/**
-	 * Remove one or more theme options.
+	 * Add a theme option.
 	 *
-	 * @param array|string $themes One or more themes to remove.
+	 * @param string $theme_option Theme name.
 	 * @return self
 	 */
-	public function remove_theme_options( $themes ): self {
+	public function add_theme_option( string $theme_option ): self {
 
-		// Convert to array if necessary.
-		if ( is_string( $themes ) ) {
-			$themes = [ $themes ];
+		// Ensure it's not already an option.
+		if ( ! $this->is_available_theme( $theme_option ) ) {
+			$this->theme_options[] = $theme_option;
 		}
 
-		// Remove as needed.
-		foreach ( $themes as $theme ) {
-			if ( $this->theme_options[ $theme ] ) {
-				unset( $this->theme_options[ $theme ] );
-			}
-		}
-
-		// Sanitize the entire thing.
-		$this->sanitize_theme_options();
-
 		return $this;
 	}
 
 	/**
-	 * Loop through the theme options, ensuring they're sanitizied.
+	 * Remove theme options.
 	 *
+	 * @param array $theme_options One or more themes names to remove.
 	 * @return self
 	 */
-	public function sanitize_theme_options(): self {
-
-		$this->theme_options = array_unique(
-			array_filter(
-				array_map(
-					function( $theme ) {
-						$theme = (string) $theme;
-						$theme = trim( $theme );
```

Comment:
Does the trimming and casting still need to occur?

Reply:
I decided to sanitize on output rather than what gets stored. It was becoming clunky otherwise, and may lead to unexpected dev experience.